lib/lz4frame.h

```diff
@@ -487,6 +487,10 @@ LZ4F_getFrameInfo(LZ4F_dctx* dctx,
  *  Use LZ4F_resetDecompressionContext() to return to clean state.
  *
  *  After a frame is fully decoded, dctx can be used again to decompress another frame.
+ *
+ *  Note: the frame header should not be passed into this function. Decompression of a frame
+ *  should start with the first byte following the frame header. Use LZ4F_headerSize() to determine
+ *  where the frame content starts.
```

Comment:
Not exactly.
It's possible to invoke `LZ4F_decompress()` directly, at the beginning of a frame.
Or at least, it should be, and if that's not the case, then there is a bug.

I suspect the issue is rather on `LZ4F_getFrameInfo()` side.
As mentioned in the documentation of this method: 
```
Its usage is optional: user can also invoke LZ4F_decompress() directly.
```

So that's the condition: `LZ4F_decompress()` can be called at the beginning of a frame,
but in this case, the optional method `LZ4F_getFrameInfo()` must not have been used.

The reason is : `LZ4F_getFrameInfo()` is effectively _starting_ the decompression process, it's a part of it. And as a consequence, it's _consuming_ the header : 
```
The number of bytes consumed from srcBuffer will be updated in *srcSizePtr (necessarily <= original value).
LZ4F_getFrameInfo() only consumes bytes when decoding has not yet started,
and when decoding the header has been successful.
```

This is in contrast with `zstd`, where the equivalent `ZSTD_getFrameHeader()` only _reads_ the header, but does not start the decompression process.

Reply:
Ya, the bug we got was from missing this bit of documentation: https://github.com/lz4/lz4/blob/073ad37f1fe1f628b1b3171410088a50f3ac165f/lib/lz4frame.h#L448
We called `LZ4F_getFrameHeader()` and then tried to decompress from the beginning of the frame. 

Do you think it would be useful to add a note about this in the `LZ4F_decompress()` docs as well? To help out poor souls like myself who failed to check the `LZ4F_getFrameHeader()` docs properly :)

Reply:
Sure !